CE-521 - Revamp API gateway documentation (#1790)
hashicorp/web-unified-docs: content/consul/v1.22.x/content/docs/north-south/api-gateway/vm/index.mdx

- [Deploy API gateway routes to VMs](/consul/docs/north-south/api-gateway/vm/route)
- [Use JWTs to verify requests to API gateways on VMs](/consul/docs/north-south/api-gateway/secure-traffic/jwt/vm)

### Reference documentation

To use Consul API gateway, you must configure the following configuration entries.

| Configuration | Description |
| ------------- | ----------- |
| [`api-gateway`](/consul/docs/reference/config-entry/api-gateway)                                      | Defines the main infrastructure resource for declaring an API gateway and listeners on the gateway. |
| [`http-route`](/consul/docs/reference/config-entry/http-route)                                        | Enables HTTP traffic to reach services in the mesh from a listener on the gateway.| 
| [`tcp-route`](/consul/docs/reference/config-entry/tcp-route)                                          | Enables TCP traffic to reach services in the mesh from a listener on the gateway.| 
| <nobr>[`file-system-certificate`](/consul/docs/reference/config-entry/file-system-certificate)</nobr> | Provides gateway with a CA certificate so that requests between the user and the gateway endpoint are encrypted. | 

> | <nobr>[`inline-certificate`](/consul/docs/reference/config-entry/inline-certificate)</nobr>           | Provides gateway with a CA certificate so that requests between the user and the gateway endpoint are encrypted. | 

Reviewer: ```suggestion
| <nobr>[`file-system-certificate`](/consul/docs/reference/config-entry/file-system-certificate)</nobr> | Provides gateway with a CA certificate at a local file path so that requests between the user and the gateway endpoint are encrypted. | 
| <nobr>[`inline-certificate`](/consul/docs/reference/config-entry/inline-certificate)</nobr>           | Provides gateway with a CA certificate directly in the configuration so that requests between the user and the gateway endpoint are encrypted. | 
```
Just drawing out the difference a little bit

---
layout: docs
page_title: API gateways on VMs overview
description: API gateways provide an ingress point for service mesh traffic. Learn how API gateways add listeners for external traffic and route HTTP requests to services in the mesh.
---

# API gateways on VMs overview

This topic provides overview information about API gateways for Consul on virtual machines. 

## Introduction

API gateways enable external network clients to access applications and services running in a Consul datacenter. Consul API gateways can also forward requests from clients to specific destinations based on path or request protocol. Systems that access services in the mesh may be internal or external to your organizational network. _North-south traffic_ is a common term to describe this type of network traffic.

## Workflow

> In order to configure an API gateway on Consul you need the following steps:

Reviewer: ```suggestion
To fully enable Consul API gateway on VMs, complete the following steps:
```

> 1. [Configure traffic encryption](#configure-traffic-encryption)

Reviewer: ```suggestion
1. [Configure gateway traffic encryption](#configure-traffic-encryption) so that the API gateway verifies the request before forwarding it to the service.
```

> 1. [Configure API gateway listeners](#configure-api-gateway-listeners)

Reviewer: ```suggestion
1. [Configure API gateway listeners](#configure-api-gateway-listeners), including the port to listen on and the encrypted certificate or token the listener requires.
```

> 1. [Configure API gateway routes](#configure-api-gateway-routes)

Reviewer: ```suggestion
1. [Define the routes](/consul/docs/north-south/api-gateway/vm/route) from the API gateway listener to the services in the mesh.
```

> 1. [Configure intentions for the API gateway](#configure-intentions-for-the-api-gateway)

Reviewer: ```suggestion
1. [Add service intentions that allow the API gateway](/consul/docs/secure-mesh/intention) to send traffic to services in the mesh. For an example configuration, refer to the [service intention configuration reference](/consul/docs/reference/config-entry/service-intentions#examples).
```

## Configure traffic encryption 

API gateways always encrypt the traffic between the external client and the exposed services.

You can encrypt the traffic with:

- [TLS certificates](/consul/docs/north-south/api-gateway/secure-traffic/encrypt).
- [JSON web tokens (JWT)](/consul/docs/north-south/api-gateway/secure-traffic/jwt/vm). <EnterpriseAlert inline />

## Configure API gateway listeners

API gateways have one or more listeners that serve as ingress points for requests to services in a Consul service mesh.

Each listener can configure the following fields:
 - `Port`, that specifies the port that the listener receives traffic on.
 - `Protocol`, that could be either `tcp` or `http`.
 - `TLS` or `JWT` configuration parameters.

Refer to [Deploy API gateway listeners to virtual machines](/consul/docs/north-south/api-gateway/vm/listener) for guidance.

## Configure API gateway routes

Routes are rule-based configurations that allow external clients to send requests to services in the mesh.

Refer to [Define API gateway routes on virtual machines](/consul/docs/north-south/api-gateway/vm/route) for guidance.

> 

Reviewer: ```suggestion
```
